pkg/contracts/solana/gateway_message.go

```diff
@@ -108,6 +116,140 @@ func (msg *MsgWithdraw) Signer() (common.Address, error) {
 	return RecoverSigner(msgHash[:], msgSig[:])
 }
 
+// MsgExecute is the message for the Solana gateway execute instruction
+type MsgExecute struct {
+	// chainID is the chain ID of Solana chain
+	chainID uint64
+
+	// Nonce is the nonce for the execute
+	nonce uint64
+
+	// amount is the lamports amount for the execute
+	amount uint64
+
+	// To is the recipient address for the execute
+	to solana.PublicKey
+
+	// signature is the signature of the message
+	signature [65]byte
+
+	// Sender is the sender address for the execute
+	sender [20]byte
+
+	// Data for execute
+	data []byte
+
+	// Remaining accounts for arbirtrary program
+	remainingAccounts []*solana.AccountMeta
+}
+
+// NewMsgExecute returns a new execute message
+func NewMsgExecute(
+	chainID, nonce, amount uint64,
+	to solana.PublicKey,
+	sender [20]byte,
+	data []byte,
+	remainingAccounts []*solana.AccountMeta,
+) *MsgExecute {
+	return &MsgExecute{
+		chainID:           chainID,
+		nonce:             nonce,
+		amount:            amount,
+		to:                to,
+		sender:            sender,
+		data:              data,
+		remainingAccounts: remainingAccounts,
+	}
+}
+
+// ChainID returns the chain ID of the message
+func (msg *MsgExecute) ChainID() uint64 {
+	return msg.chainID
+}
+
+// Nonce returns the nonce of the message
+func (msg *MsgExecute) Nonce() uint64 {
+	return msg.nonce
+}
+
+// Amount returns the amount of the message
+func (msg *MsgExecute) Amount() uint64 {
+	return msg.amount
+}
+
+// To returns the recipient address of the message
+func (msg *MsgExecute) To() solana.PublicKey {
+	return msg.to
+}
+
+// Sender returns the sender address of the message
+func (msg *MsgExecute) Sender() [20]byte {
+	return msg.sender
+}
+
+// Data returns the data of the message
+func (msg *MsgExecute) Data() []byte {
+	return msg.data
+}
+
+// RemainingAccounts returns the remaining accounts of the message
+func (msg *MsgExecute) RemainingAccounts() []*solana.AccountMeta {
+	return msg.remainingAccounts
+}
+
+// Hash packs the execute message and computes the hash
+func (msg *MsgExecute) Hash() [32]byte {
+	var message []byte
+	buff := make([]byte, 8)
+
+	message = append(message, []byte("ZETACHAIN")...)
```

Comment:
It would be good if we can define `"ZETACHAIN"` as a constant instruction identifier.